classes/order/OrderHistory.php

```diff
@@ -375,11 +375,11 @@ public function changeIdOrderState($new_order_state, $id_order, $use_existing_pa
 
         // set orders as paid
         if ($new_os->paid == 1) {
-            $invoices = $order->getInvoicesCollection();
             if ($order->total_paid != 0) {
                 $payment_method = Module::getInstanceByName($order->module);
             }
 
+            $invoices = $order->getInvoicesCollection();
             foreach ($invoices as $invoice) {
                 /** @var OrderInvoice $invoice */
                 $rest_paid = $invoice->getRestPaid();
```

Comment:
A question : what is RestPaid ? Is it the amount left to pay or the amount paid for the invoice ?

Reply:
I don't know. I think it is the amount left to pay